build-scripts/multidirectory.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+# Creates/removes/updates branches according the `jobs/` directory on the
+# master branch.  Local files and branches will be modified, so this script
+# should be executed on a clean, dedicated clone.
+set -eu
+
+set_difference() {
+    printf '%s\n%s\n%s\n' "$1" "$1" "$2" | sort | uniq -u
+}
+
+list_branches() {
+    git branch --list --remotes 'origin/*' \
```

Comment:
There's a lot going on here, mainly because there isn't a good way to list
branches that's easy for a script to parse (at least that I could find):

- we list all remote branches that match the pattern `origin/*`
- the awk pattern (`$1 ~ /…/`) filters out branches that don't have a `/` in
  their names
- `gensub(…)` removes the `origin/` prefix
- we take out `git branch`'s superfluous information (such as those two stupid
  spaces at the beginning of each line, and the `-> origin/master` you
  mentioned) by doing the substitution on `$1` instead of `$0`.

I just realized the restriction "only branches that have a `/` in their names"
can be embedded in the pattern for `git branch`. That'll make things simpler:

```
git branch --list --remotes 'origin/*/*' \
    | awk '{print gensub("^origin/", "", 1, $1)}'
```

In a perfect world, that'd be:

```
git give-me-the-list-of-branches-from-this-remote-without-extra-junk \
    origin '*/*'
```